### ISSUE TITLE
Authentication did not originate on this server

### ISSUE THREAD

**Issue by feed4rz** [almatrass/passport-opskins#5]
Getting this error when strategy is used on top of clusterized app.

I get why this kind of check exists, but it limits the functionality to 1-core app, which makes it perform less efficient. 

The solution would be to use saving/loading associations mechanism used in passport-steam strategy.

**Comment by almatrass** [comment]
Hey,
I didn't really think about this. Easiest solution I can think of is storing the codes in a JSON file instead of that array. I can do it when I get a chance but feel free to add a pull request if you have the solution made already.

**Comment by feed4rz** [comment]
I think users should be able to choose their own storage. 

Again saving/loading associations in passport-steam is a great way of doing it. Strategy gives you access to saveAssociation and loadAssociation methods in which you can defined your own way of dealing with different states.

saveAssociation is fired when there is a new state create (ex. new user is authenticating) and typically we save this state in database or somewhere else.

loadAssociation is fired when strategy needs to get a certain state to verify its origin. So in this case we do a SELECT in our database and pass state to callback or something.

We can try doing the same thing. I will try to send a PR when i ll have some free time or maybe you could try doing it yourself.

**Comment by almatrass** [comment]
Yes I'm not too familiar with the inner workings of passport itself I just kinda made this to get a site going with OPSkins login. I'll try to update it with those methods you mentioned when I get home.
